addons/stock_account/stock_account.py

```diff
@@ -24,6 +24,80 @@ def post_inventory(self, cr, uid, inv, context=None):
             ctx['force_period_date'] = inv.accounting_date
         return super(stock_inventory, self).post_inventory(cr, uid, inv, context=ctx)
 
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _anglo_saxon_sale_move_lines(self, name, product, uom, qty, price_unit, currency=False, amount_currency=False, fiscal_position=False, account_analytic=False):
+
+        if product.type == 'product' and product.valuation == 'real_time':
+            accounts = product.product_tmpl_id.get_product_accounts(fiscal_pos=fiscal_position)
+            # debit account dacc will be the output account
+            dacc = accounts['stock_output'].id
+            # credit account cacc will be the expense account
+            cacc = accounts['expense'].id
+            if dacc and cacc:
+                return [
+                    {
+                        'type': 'src',
+                        'name': name[:64],
+                        'price_unit': price_unit,
+                        'quantity': qty,
+                        'price': price_unit * qty,
+                        'currency_id': currency and currency.id,
+                        'amount_currency': amount_currency,
+                        'account_id':dacc,
+                        'product_id':product.id,
+                        'uom_id':uom.id,
+                        'account_analytic_id': account_analytic and account_analytic.id,
+                    },
+
+                    {
+                        'type': 'src',
+                        'name': name[:64],
+                        'price_unit': price_unit,
+                        'quantity': qty,
+                        'price': -1 * price_unit * qty,
+                        'currency_id': currency and currency.id,
+                        'amount_currency': -1 * amount_currency,
+                        'account_id':cacc,
+                        'product_id':product.id,
+                        'uom_id':uom.id,
+                        'account_analytic_id': account_analytic and account_analytic.id,
+                    },
+                ]
+        return []
+
+    @api.model
+    def line_get_convert(self, line, partner):
```

Comment:
The original method from account.invoice should call this one, instead of duplicating the code + mark the old one as deprecated (to be removed in master).
Also the naming is bad, we could take this opportunity to name the new one better and make it private, .e.g. `_convert_prepared_anglosaxon_line()`